crates/sui-core/src/consensus_handler.rs

```diff
@@ -293,6 +294,19 @@ impl<T: ParentSync + Send + Sync> ExecutionState for ConsensusHandler<T> {
             }
         }
 
+        // TODO: make the reordering algorithm richer and depend on object hotness as well.
+        // Order transactions based on their gas prices. System transactions without gas price
+        // are put to the beginning of the sequenced_transactions vector.
+        if matches!(
+            self.epoch_store
+                .protocol_config()
+                .consensus_transaction_ordering(),
+            ConsensusTransactionOrdering::ByGasPrice
+        ) {
+            let _scope = monitored_scope("HandleConsensusOutput::order_by_gas_price");
+            order_by_gas_price(&mut sequenced_transactions);
```

Comment:
What is the typical and max size of sequenced_transactions?

Reply:
The size of `sequenced_transactions` is the number of Sui transactions in a Narwhal commit + 1 (prologue). In 100 node deployments, Narwhal commits happen every 1~2s. Actual size of `sequenced_transactions` is correlated with TPS. The size limit is much larger based on Narwhal limits.

Reply:
If we are concerned about added latency due to this additional step, I can run an experiment in private testnet with high TPS and observe what the monitored scope metrics show us.

Reply:
I don't think we're going to have any problems caused by sorting - consensus handler is dominated by db reads and writes. Anything that is purely in memory like this should not cause problems